shtdlib.sh

```diff
@@ -1914,6 +1914,14 @@ function uri_unparser {
     echo "${working_uri}"
 }
 
+function uri_hostname_to_fqdn {
```

Comment:
Suggest adding a comment above to document usage

Reply:
added

Reply:
// Start pedantic rant //
Your comment is going to be  read by someone figuring out if this is a function they want/need to use, it should be written for them with that in mind. Given that I'd suggest some changes for clarity.

Perhaps something like:
Opportunistically resolves the hostname portion of an URI and replaces it with the fully qualified hostname (FQDN) using hosts file and DNS servers and search domains specified in resolv.conf. If no match can 

This also raises some questions:

- It's also not clear how/what is returned, perhaps the function should echo/print the result for ease of use rather than leaving it in an "undocumented" variable?
- What happens if there is no match, does it return the unresolved hostname, empty string, raise an error, should it return a non 0 status when no match is found?

// End pedantic rant //